homeassistant/components/kdeconnect/storage.py

```diff
@@ -0,0 +1,108 @@
+"""Implementation of a storage to store data for KDE Connect."""
+from __future__ import annotations
+
+from pathlib import Path
+
+from cryptography.hazmat.primitives.asymmetric.rsa import RSAPrivateKey
+from cryptography.x509 import Certificate, load_pem_x509_certificate
+from pykdeconnect.devices import KdeConnectDevice
+from pykdeconnect.helpers import CertificateHelper
+from pykdeconnect.storage import AbstractStorage
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_DEVICE_ID
+from homeassistant.core import HomeAssistant
+
+from .const import (
+    CONF_DEVICE_CERT,
+    CONF_DEVICE_INCOMING_CAPS,
+    CONF_DEVICE_NAME,
+    CONF_DEVICE_OUTGOING_CAPS,
+    CONF_DEVICE_TYPE,
+    DOMAIN,
+)
+
+
+class HomeAssistantStorage(AbstractStorage):
+    """A storage for home assistant."""
+
+    hass: HomeAssistant
+    _devices: dict[str, KdeConnectDevice]
+    _device_id: str
+
+    private_key: RSAPrivateKey
+    cert: Certificate
```

Comment:
`private_key` and `cert` are never referenced outside of the constructor so they can be removed

Reply:
You could also consider moving the key setup into a separate function invoked from the init methods